tests/test_web_api.py

```diff
@@ -0,0 +1,307 @@
+        self.assertEqual(204, response.status_code)
+
+        response = self.client.get(url.path, format="json")
+        self.assertEqual(404, response.status_code)
+
+        response = self.client.get("/rp/job/pipeline/", format="json")
+        self.assertEqual(0, len(response.data))
+
+
+class RegulationFileTestCase(APITestCase):
+
+    file_contents = "123"
+    hashed_contents = None
+
+    def get_hashed_contents(self):
+        if self.hashed_contents is None:
+            try:
+                self.hashed_contents = md5(self.file_contents.encode(
+                    "utf-8")).hexdigest()
+            except AttributeError:
+                self.hashed_contents = md5(self.file_contents).hexdigest()
+        return self.hashed_contents
+
+    def test_create_file(self):
+        with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+            tmp.write(self.file_contents.encode("utf-8"))
+            tmp_name = ospath.split(tmp.name)[-1]
+            tmp.seek(0)
+            response = self.client.post(
+                "/rp/job/upload/", {"file": tmp})
+        self.assertEquals(201, response.status_code)
+        data = response.data
+        self.assertEquals(self.get_hashed_contents(), data["hexhash"])
+        self.assertEquals(tmp_name, data["filename"])
+        self.assertEquals("File contents not shown.", data["contents"])
+        self.assertEquals(file_url(self.get_hashed_contents()), data["url"])
+        return response
+
+    def test_reject_duplicates(self):
+        self.test_create_file()
+        with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+            tmp.write(self.file_contents.encode("utf-8"))
+            tmp.seek(0)
+            response = self.client.post(
+                "/rp/job/upload/", {"file": tmp})
+        self.assertEquals(400, response.status_code)
+        self.assertIn("error", response.data)
+        self.assertEquals("File already present.", response.data["error"])
+
+    def test_reject_large(self):
+        with patch("regparser.web.jobs.views.FileUploadView",
+                   new=PatchedFileUploadView) as p:
+            p.size_limit = 10000
+            with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+                tmp.write(self.file_contents.encode("utf-8"))
+                tmp.seek(0)
+                response = self.client.post(
+                    "/rp/job/upload/", {"file": tmp})
+            self.assertEquals(201, response.status_code)
+
+            with NamedTemporaryFile(suffix=".xml", delete=True) as tmp:
+                contents = "123" * 100001
+                tmp.write(contents.encode("utf-8"))
+                tmp.seek(0)
+                response = self.client.post(
+                    "/rp/job/upload/", {"file": tmp})
+            self.assertEquals(400, response.status_code)
+            self.assertEquals("File too large (10000-byte limit).",
+                              response.data["error"])
```

Comment:
No biggie, but why not make this much smaller, like 10 bytes? As it stands, this test needs to write a 300K file every time it runs, which seems excessive